src/main/scala/io/iohk/ethereum/blockchain/sync/SyncStateSchedulerActor.scala

```diff
@@ -43,51 +86,60 @@ class SyncStateSchedulerActor(downloader: ActorRef, sync: SyncStateScheduler, sy
-          s"was ${result.error}"
-      )
-      lastReceivedCommand match {
-        case Some((startSignal: StartSyncingTo, sender)) =>
-          val initStats = ProcessingStatistics().addSaved(result.writtenElements)
-          val initState = startSyncing(startSignal.stateRoot, startSignal.blockNumber)
-          context become (syncing(initState, initStats, startSignal.blockNumber, sender))
-        case Some((restartSignal: RestartRequested.type, sender)) =>
-          sender ! WaitingForNewTargetBlock
-          context.become(idle(ProcessingStatistics().addSaved(result.writtenElements)))
-        case _ =>
-          context.become(idle(ProcessingStatistics().addSaved(result.writtenElements)))
-      }
-
-    case command: SyncStateSchedulerActorCommand =>
-      context.become(waitingForBloomFilterToLoad(Some((command, sender()))))
-  }
+  def waitingForBloomFilterToLoad(lastReceivedCommand: Option[(SyncStateSchedulerActorCommand, ActorRef)]): Receive =
+    handleCommonMessages orElse {
+      case BloomFilterResult(result) =>
+        log.debug(
+          s"Loaded ${result.writtenElements} already known elements from storage to bloom filter the error while loading " +
+            s"was ${result.error}"
+        )
+        lastReceivedCommand match {
+          case Some((startSignal: StartSyncingTo, sender)) =>
+            val initStats = ProcessingStatistics().addSaved(result.writtenElements)
+            val initState = startSyncing(startSignal.stateRoot, startSignal.blockNumber)
+            context become (syncing(
+              SyncSchedulerActorState.initial(initState, initStats, startSignal.blockNumber, sender)
+            ))
+          case Some((restartSignal: RestartRequested.type, sender)) =>
+            sender ! WaitingForNewTargetBlock
+            context.become(idle(ProcessingStatistics().addSaved(result.writtenElements)))
+          case _ =>
+            context.become(idle(ProcessingStatistics().addSaved(result.writtenElements)))
+        }
+
+      case command: SyncStateSchedulerActorCommand =>
+        context.become(waitingForBloomFilterToLoad(Some((command, sender()))))
+    }
 
   private def startSyncing(root: ByteString, bn: BigInt): SchedulerState = {
     timers.startTimerAtFixedRate(PrintInfoKey, PrintInfo, 30.seconds)
     log.info(s"Starting state sync to root ${ByteStringUtils.hash2string(root)} on block ${bn}")
     //TODO handle case when we already have root i.e state is synced up to this point
     val initState = sync.initState(root).get
-    val (initialMissing, state1) = initState.getAllMissingHashes
-    downloader ! RegisterScheduler
-    downloader ! GetMissingNodes(initialMissing)
-    state1
+    self ! Sync
+    initState
   }
 
-  def idle(processingStatistics: ProcessingStatistics): Receive = {
+  def idle(processingStatistics: ProcessingStatistics): Receive = handleCommonMessages orElse {
     case StartSyncingTo(root, bn) =>
       val state1 = startSyncing(root, bn)
```

Comment:
`startSyncing` method is always followed by `SyncSchedulerActorState.initial`. Maybe that call could be part of `startSyncing` method?